source/particle/generator/random_uniform.cc

```diff
@@ -0,0 +1,205 @@
+/*
+  Copyright (C) 2015 by the authors of the ASPECT code.
+
+ This file is part of ASPECT.
+
+ ASPECT is free software; you can redistribute it and/or modify
+ it under the terms of the GNU General Public License as published by
+ the Free Software Foundation; either version 2, or (at your option)
+ any later version.
+
+ ASPECT is distributed in the hope that it will be useful,
+ but WITHOUT ANY WARRANTY; without even the implied warranty of
+ MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ GNU General Public License for more details.
+
+ You should have received a copy of the GNU General Public License
+ along with ASPECT; see the file doc/COPYING.  If not see
+ <http://www.gnu.org/licenses/>.
+ */
+
+#include <aspect/particle/generator/random_uniform.h>
+
+#include <boost/random.hpp>
+
+
+namespace aspect
+{
+  namespace Particle
+  {
+    namespace Generator
+    {
+      template <int dim>
+      RandomUniform<dim>::RandomUniform() {}
+
+      template <int dim>
+      std::multimap<types::LevelInd, Particle<dim> >
+      RandomUniform<dim>::generate_particles()
+      {
+        // Calculate the number of particles in this domain as a fraction of total volume
+        double local_volume = 0;
+        for (typename parallel::distributed::Triangulation<dim>::active_cell_iterator
+             it=this->get_triangulation().begin_active();
+             it!=this->get_triangulation().end(); ++it)
+          {
+            const double cell_volume = it->measure();
```

Comment:
the problem with this is that you don't take into account the mapping. I think it would be useful to simply adapt the function you had for the probability density generator